Shared/ViewModels/UserSignInViewModel.swift

```diff
@@ -73,59 +63,176 @@ final class UserSignInViewModel: ViewModel {
+
+    /// We don't use Timer.scheduledTimer because checking quick connect status is async. We only want to start the next
+    /// poll when the current has finished.
+    private var quickConnectPollTask: Task<Void, Never>?
+    /// When two quick connect tasks are started for whatever reason, cancelling the repeating task above seems to fail
+    /// and the attempted cancelled task seems to continue to spawn more repeating tasks. We ensure only a single
+    /// poll task continues to live with this ID.
+    private var quickConnectPollTaskID: UUID?
+    /// If, for whatever reason, the monitor task keeps going, we don't want to let it silently run forever.
+    private let quickConnectMaxRetries = 200
+
+    enum QuickConnectStatus {
+        case fetchingSecret
+        // After secret and code is fetched from the server, store it in the associated value as:
+        //                         (secret, code)
+        case awaitingAuthentication(String, String)
+        // Store the error and surface it to user if possible
+        case fetchingSecretFailed(Error?)
+    }
 
-            guard let response else { return }
+    func startQuickConnect() -> AsyncStream<QuickConnectResult> {
+        logger.debug("Attempting to start quick connect...")
+        quickConnectStatus = .fetchingSecret
 
-            await MainActor.run {
-                quickConnectSecret = response.value.secret
-                quickConnectCode = response.value.code
+        Task {
+            let initiatePath = Paths.initiate
+            do {
+                let response = try await client.send(initiatePath)
+
+                guard let secret = response.value.secret,
+                      let code = response.value.code
+                else {
+                    // TODO: Create an error & display it in QuickConnectView (iOS)/UserSignInView (tvOS)
+                    quickConnectStatus = .fetchingSecretFailed(nil)
+                    return
+                }
+
+                await MainActor.run {
+                    quickConnectStatus = .awaitingAuthentication(secret, code)
+                }
+
+            } catch {
+                quickConnectStatus = .fetchingSecretFailed(error)
             }
         }
 
-        return .init { continuation in
+        let taskID = UUID()
+        quickConnectPollTaskID = taskID
 
-            checkAuthStatus(continuation: continuation)
+        return .init { continuation in
+            checkAuthStatus(continuation: continuation, id: taskID)
         }
     }
 
-    private func checkAuthStatus(continuation: AsyncStream<QuickConnectResult>.Continuation) {
-
+    private func checkAuthStatus(continuation: AsyncStream<QuickConnectResult>.Continuation, id: UUID, tries: Int = 0) {
         let task = Task {
-            guard let quickConnectSecret else { return }
+            // Don't race into failure while we're fetching the secret.
+            while case .fetchingSecret = quickConnectStatus {
```

Comment:
Why is this necessary? We should always have the secret before checking auth.

Reply:
This was actually what was causing the bug: we _didn't_ have the secret before `checkAuthStatus`, so the poll loop would immediately terminate. A typical failure case would look like:
1. Call `startQuickConnect`
2. Initiate fetching secret
3. Call `checkAuthStatus`, which terminates immediately because secret isn't fetched